scripts/o3de/ExportScripts/export_standalone_monolithic_engine_centric.py

```diff
@@ -0,0 +1,160 @@
+parser.add_argument('-bnmt', '--build-non-mono-tools', action='store_true')
+parser.add_argument('-nmbp', '--non-mono-build-path', type=pathlib.Path, default=None)
+parser.add_argument('-mbp', '--mono-build-path', type=pathlib.Path, default=None)
+parser.add_argument('-nogame', '--no-game-launcher', action='store_true', help='this flag skips building the Game Launcher on a platform if not needed.')
+parser.add_argument('-noserver', '--no-server-launcher', action='store_true', help='this flag skips building the Server Launcher on a platform if not needed.')
+parser.add_argument('-nounified', '--no-unified-launcher', action='store_true', help='this flag skips building the Unified Launcher on a platform if not needed.')
+parser.add_argument('-pl', '--platform', type=str, default=host_asset_platform, choices=['pc', 'linux', 'mac'])
+parser.add_argument('-a', '--archive-output', action='store_true', help='This option places the final output of the build into a compressed archive')
+parser.add_argument('-q', '--quiet', action='store_true', help='Suppresses logging information unless an error occurs.')
+
+args = parser.parse_args(o3de_context.args)
+
+if args.quiet:
+    o3de_logger.setLevel(logging.ERROR)
+
+non_mono_build_path = (o3de_context.engine_path) / 'build' / 'non_mono' if args.non_mono_build_path is None else args.non_mono_build_path
+mono_build_path     = (o3de_context.engine_path) / 'build' / 'mono' if args.mono_build_path is None else args.mono_build_path
+
+selected_platform = args.platform
+
+if not selected_platform:
+    o3de_logger.error("Unable to identify default host platform! Please supply the platform using '--platform'. Options are [pc, linux, mac].")
+    sys.exit(1)
+
+o3de_logger.info(f"Selected Project path: {o3de_context.project_path}")
+o3de_logger.info(f"Selected Engine path : {o3de_context.engine_path}")
+o3de_logger.info(f"Build path for non-monolithic executables: {args.non_mono_build_path}")
+o3de_logger.info(f"Build path for monolithic executables: {args.mono_build_path}")
+
+output_cache_path = args.output_path / 'Cache' / selected_platform
+os.makedirs(output_cache_path, exist_ok=True)
+
+# Build project assets and the engine (non-monolithic)
+if args.build_non_mono_tools:
+    process_command(['cmake', '-S', o3de_context.engine_path, '-B', str(non_mono_build_path), '-DLY_MONOLITHIC_GAME=0', f'-DLY_PROJECTS={o3de_context.project_path}'],
+                    cwd=o3de_context.engine_path)
+    process_command(['cmake', '--build', str(non_mono_build_path), '--target', 'AssetBundler', 'AssetBundlerBatch', 'AssetProcessor', 'AssetProcessorBatch', '--config','profile'],
+                    cwd=o3de_context.engine_path)
+    process_command(['cmake', '--build', str(non_mono_build_path), '--target', f'{project_name}.Assets', '--config', 'profile'],
+                    cwd=o3de_context.engine_path)
```

Comment:
You shouldn't need do do this in 2 different cmake commands, you should be able to specify all the targets including`${project_name}.Assets`. (And also, you probably don't need `AssetProcessor`. `AssetProcessorBatch` should be enough to compile the assets.
```suggestion
    process_command(['cmake', '--build', str(non_mono_build_path), '--target', 'AssetBundler', 'AssetBundlerBatch', 'AssetProcessor', 'AssetProcessorBatch', f'{project_name}.Assets', '--config','profile'],
                    cwd=o3de_context.engine_path)
```

Reply:
This came up during development for MPS export script as well. The main reason I chose to split it up this way was to make troubleshooting easier. Doing it this way makes it easier to trace logs whether building assets failed, or building the executables failed, as each command is isolated.

Note taken on `AssetProcessor`. I can remove that.

Reply:
Probably don't need `AssetBundler` either